packages/nvidia_nat_a365/tests/front_end/test_integration.py

```
@@ -0,0 +1,1425 @@
+        """Test that missing notifications package is handled gracefully."""
+        # Remove the module from sys.modules to force re-import, then patch __import__ to fail
+        import sys
+        original_notifications = sys.modules.pop("microsoft_agents_a365.notifications", None)
+        original_models = sys.modules.pop("microsoft_agents_a365.notifications.models", None)
+
+        # Patch __import__ to raise ModuleNotFoundError for this specific import.
+        # ModuleNotFoundError is what Python actually raises when a package is missing,
+        # and is what setup_notification_handlers narrowly catches for graceful degradation.
+        original_import = __import__
+
+        def mock_import(name, globals=None, locals=None, fromlist=(), level=0):
```

Ruff (0.15.12)

[error] Function argument `globals` is shadowing a Python builtin

(A002)